### ISSUE TITLE
model.fit very slow validation, more evident with Intel Phi

### ISSUE THREAD

**Issue by ftarlao** [keras-team/keras#11267]
I have a similar problem of this old [issue](https://github.com/keras-team/keras/issues/7723)(wrongly closed because stale), in all epochs, the validation calculation is slower than the epoch training part.
Both validation and training sets have the same number of examples.

The two phases take the same time with a 18 cores Xeon CPU, but validation takes 4 times the training time with Intel Phi architeclure (and tensorflow MKL binary).

I think/suspect that model evaluation for validation calculation does not take advantage of parallelization/multi-threading. This could be the core of the problem. Please check.

Regards

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] THE LINKED OLD ISSUE ALREADY HAS IT. Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by caseybasichis** [comment]
I’m encountering the same issue on GPU with a MSE loss.  At least 4x as long to validate 1/10th of the samples used in training.
